project1/client/client_tasks/download_task.cpp

```diff
@@ -0,0 +1,18 @@
+#include "download_task.h"
+
+namespace client_tasks {
+thread_pool::Task::Status DownloadTask::RunAtomic() {
+  while(!parser_.HasCompleteMessage()) {
+    incoming_file_buf_.resize(buffer_size_);
+    recv(client_fd_, incoming_file_buf_.data(), buffer_size_, 0);
+    incoming_file_buf_.resize(buffer_size_);
```

Comment:
I think what you mean to do is check the return value of `recv` and resize the buffer to this value. You probably should do that anyway for error-handling purposes...